### PR TITLE
Resource per metric

### PR DESCRIPTION
**Pull request by rghetia**
added test for 
- per metric resource
- built-in metrics
- metric prefix.

### CODE REVIEW

File: metrics_proto_api_test.go [hunk 1]
```diff
@@ -85,6 +86,74 @@ func TestVariousCasesFromFile(t *testing.T) {
 	}
 }
 
+func TestMetricsWithPrefix(t *testing.T) {
+	server, addr, doneFn := createFakeServer(t)
```
Comment (songy23) [inline metrics_proto_api_test.go]: Nit: looks like these few lines appear multiple times. How about having a func `createFakeServerAndConn` that returns both the server and connection? The returned `doneFn` can incorporate `conn.Close()` too.

File: metrics_proto_api_test.go [hunk 2]
```diff
@@ -85,6 +86,74 @@ func TestVariousCasesFromFile(t *testing.T) {
...
+	defer doneFn()
+
+	// Now create a gRPC connection to the server.
+	conn := createConn(t, addr)
+	defer conn.Close()
+
+	tc := readTestCaseFromFiles(t, "SingleMetric")
+
+	prefixes := []string{
```
Comment (songy23) [inline metrics_proto_api_test.go]: Nit: can also test "external.googleapis.com/prometheus/".

File: metrics_proto_api_test.go [hunk 3]
```diff
@@ -405,12 +534,15 @@ func createFakeServer(t *testing.T) (*fakeMetricsServer, string, func()) {
 	go func() {
 		_ = srv.Serve(ln)
 	}()
+	_, serverPortStr, _ := net.SplitHostPort(ln.Addr().String())
+	conn := createConn(t, "localhost:"+serverPortStr)
+
 	stop := func() {
 		srv.Stop()
 		_ = ln.Close()
+		conn.Close()
```
Comment (songy23) [inline metrics_proto_api_test.go]: Nit: I think `conn` needs to be closed first.

Comment (rghetia) [reply metrics_proto_api_test.go]: yes. fixed it.

**Pull request merged**